src/vm/internal/otable.cc

```diff
@@ -1,271 +1,129 @@
+#ifndef TESTING
```

Comment:
why do you need different include files for TESTING?

Reply:
Probably unnecessary just for convenience since I wanted a simple one liner to compile the test code.

Reply:
Actually I looked at this again the include files actually do need to be different because of showStatus and object_t*. For unit testing i don't want to link in object.o and all the files it might depend on.

Reply:
You don't have to link object.o for the struct definition, just include the appropriate headers .

Reply:
and in fact if the code moves to a more C++ style where the header doesn't just include forward declarations and structs. One is forced to link in the various files as well which seems to some extent defeat the purpose of having the units be independent.

Reply:
include headers doesn't force you to link anything.  The current way is simply wrong.  

What you should do , and how unit test should work, is that it allows you to compile otable.o and otable_test.o , and link those together to run.   otable.o should not have any "TESTING" define , you have to figure out how to do that.